content/HTTPS is not enough - Pablo Blanco.md

```diff
@@ -0,0 +1,218 @@
+
+If a site is listed, the browser always connects to it using HTTPS, even on the
+first visit. This effectively eliminates the only significant vulnerability in
+HTTPS enforcement, ensuring end-to-end encryption.
+
+**To add our site to the preloaded list we need:**
+
+1. Present a valid certificate
+2. Redirect all HTTP requests on port 80 to HTTPS
+3. Include the HSTS header in any additional HTTPS redirects
+4. Serve all subdomains over HTTPS
+5. Specify all header directives: max-age must be at least 31536000 seconds (1
+   year), and includesubdomains and preload must be present
+
+### Content Security Policy (CSP)
+
+Another header that can help us is Content Security Policy (CSP), this is used
+to detect and mitigate certain types of attacks like cross-site scripting (XSS),
+clickjacking and other code injection attacks resulting from execution of
+malicious content in the trusted web page context. A CSP can be used for simple
+purposes like enforcing HTTPS on SSL-enabled sites, to more sophisticated uses
+like authorizing only truly trusted sources and blocking others.
+
+Using CSP also makes sure that we don’t have mixed content errors on your site.
+For example, if we have a HTTPS site and HTTP is embedded in the site (like a
+video), then we will wind up with mixed content errors in the browser console. A
+CSP could direct a website to make all URLs are always HTTPS, to avoid any mixed
+content errors.
+
+This header contains only the policy, and can be defined this way:
+
+`Content-Security-Policy: default-src 'self'`
+
+Here we are saying that all content to come from the site's own origin
+(excluding subdomains.) The policy provides many options that can be found
+[here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
+
+**How can I check if any site uses security headers?**
+
+We can check server response headers by using curl or nmap (more detailed
+information), here a few examples:
+
+**_An example of a redirection without HSTS or CSP_**
+
+```
+$> curl --head https://www.not-hsts-header.com
+HTTP/1.1 301 Moved Permanently
+Content-length: 0
+```
+
+```
+$> nmap -p 443 --script http-security-headers not-hsts-header.com
+....
+PORT    STATE SERVICE
+443/tcp open  https
+| http-security-headers:
+|   Strict_Transport_Security:
+|     HSTS not configured in HTTPS Server
+|   X_XSS_Protection:
+|     Header: X-XSS-Protection: 0
+|     Description: The XSS filter is disabled.|
+|_
+```
+
+**_An example with HSTS_**
+
+```
+$> curl --head https://www.hsts-header-enabled.com
+HTTP/2 200
+strict-transport-security: max-age=15552000; preload
+```
+
+```
+$> nmap -p 443 --script http-security-headers hsts-header-enabled.com
+....
+PORT    STATE SERVICE
+443/tcp open  https
+| http-security-headers:
+|   Strict_Transport_Security:
+|     Header: Strict-Transport-Security: max-age=15552000; preload
+|   X_XSS_Protection:
+|     Header: X-XSS-Protection: 0
+|     Description: The XSS filter is disabled.
+|_
+```
```

Comment:
What do you think about putting these examples on gists?

Reply:
Yes, I already tested with gists but I am not going to apply them for this case, I prefer something more plain for bash commands, like this one:

<img width="809" alt="Screen Shot 2020-06-30 at 10 29 36" src="https://user-images.githubusercontent.com/3788239/86132428-3f314400-babd-11ea-98d6-c90d6af9117d.png">

Anyways I can include gists link on the footer of each one.

Reply:
got it, yeah, I wasn't sure if it would make sense either because they were bash, we can check how they look like on the post later 👍